src/Microsoft.VisualStudio.Composition/Resolver.cs

```diff
@@ -80,18 +80,15 @@ private StrongAssemblyIdentity NotifyAssemblyLoaded(Assembly assembly, AssemblyN
                 assemblyName = assembly.GetName();
             }
 
-            lock (this.loadedAssemblyStrongIdentities)
+            if (this.TryGetAssemblyId(assemblyName, out StrongAssemblyIdentity result))
             {
-                if (this.loadedAssemblyStrongIdentities.TryGetValue(assemblyName, out var result))
-                {
-                    return result;
-                }
+                return result;
             }
 
             var assemblyId = StrongAssemblyIdentity.CreateFrom(assembly, assemblyName);
             lock (this.loadedAssemblyStrongIdentities)
             {
-                if (!this.loadedAssemblyStrongIdentities.TryGetValue(assemblyName, out var result))
+                if (!this.loadedAssemblyStrongIdentities.TryGetValue(assemblyName, out result))
```

Comment:
I'm not too familiar with the code here - why not use TryGetAssemblyId here as well?

Reply:
Because this isn't just a check within a lock: it's a check-then-initialize multi-step that all has to be within a lock. So calling the other method would not take a lock with a broad enough scope, resulting in two locks, which while still cheap, makes the helper method also worth next to nothing.

Reply:
Makes sense to me, thanks!